src/main/java/com/pinterest/secor/common/KafkaClient.java

```diff
@@ -127,8 +129,13 @@ private Message getMessage(TopicPartition topicPartition, long offset,
         FetchResponse response = consumer.fetch(request);
         if (response.hasError()) {
             consumer.close();
-            throw new RuntimeException("Error fetching offset data. Reason: " +
-                    response.errorCode(topicPartition.getTopic(), topicPartition.getPartition()));
+            int errorCode = response.errorCode(topicPartition.getTopic(), topicPartition.getPartition());
+
+            if (errorCode == 1) {
```

Comment:
See response codes from the Kafka docs:

https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-PossibleErrorCodes.1